docs/add-ons/configuration.md

```diff
@@ -87,7 +87,7 @@ We support the following build arguments by default:
 
 ## Add-on config
 
-The configuration for an add-on is stored in `config.(json/yaml/yml)`.
+The configuration for an add-on is stored in `config.(json/yaml/yml)`. Avoid the use of this filename for anything other than add-on configuration, as the Supervisor does a recursive lookup.
```

Comment:
It will halt for the first found occurrence in a tree?

Reply:
I would expect that to be the case, but it is not my experience. It is based on the reply I got from the issue I opened for Supervisor. Note that I have not looked at the code, so I don't know what is intended.

It could be that I misunderstood the reply in the other issue, or that it is a bug in Supervisor, but if I add a repository with the following tree structure to the store (simplified, please see the other issue if you have time) I get a warning in the supervisor store log about invalid Add-on config in `subfolder1/subfolder2/config.yaml`.
```
<repository root>
  repository.json
  addon1/
    config.json
    subfolder1/
      subfolder2/
        config.yaml
```

Reply:
FYI home-assistant/supervisor#2754

Reply:
Upstream has been marked as a bug now. I think it makes more sense to hold off until that is fixed to prevent incorrect documentation later on.